src/shared/sam/cli/samCliVersionValidator.ts

```diff
@@ -35,46 +43,14 @@ export class SamCliVersionValidator {
         'Update AWS Toolkit'
     )
 
-    private readonly _versionProvider: SamCliVersionProvider
-
-    public constructor(versionProvider?: SamCliVersionProvider) {
-        this._versionProvider = versionProvider || new DefaultSamCliVersionProvider()
-    }
-
-    public async validate(): Promise<SamCliVersionValidatorResult> {
-        const version = await this._versionProvider.getSamCliVersion()
-
+    public async getCliValidationStatus(version?: string): Promise<SamCliVersionValidatorResult> {
         return {
             version: version,
-            validation: SamCliVersion.validate(version)
+            validation: this.getValidationStatus(version)
         }
     }
 
-    /**
-     * Call this to determine the SAM CLI version, and inform the user whether or not the version is acceptable
-     */
-    public async validateAndNotify(): Promise<void> {
-        const validationResult = await this.validate()
-
-        if (validationResult.validation === SamCliVersionValidation.Valid) {
-            await this.notifyVersionIsValid(validationResult.version)
-        } else {
-            await this.notifyVersionIsNotValid(validationResult)
-        }
-    }
-
-    private async notifyVersionIsValid(version: string): Promise<void> {
-        vscode.window.setStatusBarMessage(
-            localize(
-                'AWS.samcli.notification.version.valid',
-                'Your SAM CLI version {0} is valid.',
-                version
-            ),
-            SamCliVersionValidator.STATUS_BAR_DURATION_MILLIS
-        )
-    }
-
-    private async notifyVersionIsNotValid(validationResult: SamCliVersionValidatorResult): Promise<void> {
+    public async notifyVersionIsNotValid(validationResult: SamCliVersionValidatorResult): Promise<void> {
```

Comment:
This doesn't feel like the right place to expose notification. The validator is responsible for determining what (if anything) is wrong, the caller is responsible for relaying that information to the user.

Consider either moving notification logic to the caller, or adding a new `Notifier` type to handle notification logic.

Reply:
Honestly, if we remove this, we should probably just merge this file into the invoker (since the only other functionality is turning the version into an enum).

Reply:
Haven't done the above yet but strongly considering it.